### DISCUSSION TITLE
Expected "Call from invalid thread", but it worked unexpectedly

### DISCUSSION THREAD

**Discussion by penenkel** [AvaloniaUI/Avalonia#6814] [Q&A]
Coming from WPF, I expected to receive a "Call from invalid thread" when ever a Non-UiThread attempts to update the UI. Regardless of how the update is effected, i.e. whether (1) by setting a property directly on the control, or (2) by updating a ViewModel property that is bound to a control-property, or (3) by advancing an IObservable property on the ViewModel that is ^-bound to the control.

But apparently in Avalonia it is save to do (2) or (3) from any thread? At least I did not get any exceptions there and die View updated as desired. Only when tying (1) did I get the expected error.

I just want to confirm with you that this is by design and something I can rely on holding true in future versions.

I am currently using Avalonia 0.10.7 on both WIndows and Linux.

**Comment by maxkatz6** [comment]
> But apparently in Avalonia it is save to do (2) or (3) from any thread

No. Except observable probably is handled on main thread scheduler, which makes it safe, but I am not sure. CC @grokys 

> I can rely on holding true in future versions.

You shouldn't.

It's a good question to discuss though.
